static/src/js/components/PlatformField/PlatformField.jsx

```diff
@@ -0,0 +1,249 @@
+import React, { useEffect, useState } from 'react'
+import PropTypes from 'prop-types'
+import Select from 'react-select'
+import fetchCmrKeywords from '../../utils/fetchCmrKeywords'
+import parseCmrResponse from '../../utils/parseCmrResponse'
+import './PlatformField.scss'
+
+/** This is custom field found with Collections >> acquisitionInformation
+ *
+ * @typedef {Object} PlatformField
+ * @property {Function} onChange A callback function triggered when the user inputs a text.
+ * @property {Object} uiSchema uiSchema for the field being shown.
+ * @property {Object} formData Saved Draft
+ */
+const PlatformField = ({ onChange, uiSchema, formData }) => {
+  const { Type, ShortName, LongName } = formData
+  const [type, setType] = useState(Type || '')
+  const [shortName, setShortName] = useState(ShortName || '')
+  const [longName, setLongName] = useState(LongName || '')
+  const [loading, setLoading] = useState(false)
+  const [keyword, setKeyword] = useState([])
+  const [showMenu, setShowMenu] = useState(false)
+  const [shouldFocus, setShouldFocus] = useState(false)
+  const [longNameMap, setLongNameMap] = useState({})
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const controlled = uiSchema['ui:controlled']
+      const { name, controlName } = controlled
+
+      setLoading(true)
+
+      const keywords = await fetchCmrKeywords(name)
+      const paths = parseCmrResponse(keywords, controlName)
+      const newPaths = []
+
+      // Creating a '>' delimiter map for the platform keywords.
+      // For example: if the values from the parseCmrResponse are
+      //                basis="Air-based Platform",
+      //                Category="Jet", shortName="A340-600"
+      //                LongName="Airbus A340-600"
+      // the '>' delimited path would for this case would be
+      //                Air-based Platforms>Jet>A340-600>Airbus A340-600
+      // This path is being used in the render to display the Short Names (A340-600)
+      // and based on the selected Short Name it will auto populate the Type (JET)
+      // and Long Name (Airbus A340-600) field
+
+      paths.forEach((path) => {
+        let newPath = ''
+        path.forEach((value, index) => {
+          newPath += `>${value}`
+          if (index === 3) {
+            setLongNameMap((prevMap) => ({
+              ...prevMap,
+              [path[2]]: value
+            }))
+          }
+
+          if (!newPaths.includes(newPath.slice(1))) {
+            newPaths.push(newPath.slice(1))
+          }
+        })
+      })
+
+      setLoading(false)
+      setKeyword(newPaths)
+    }
+
+    fetchData()
+  }, [])
+
+  const onHandleMouseDown = (values) => {
+    const [valueType, valueShortName] = values
+    const valueLongName = longNameMap[valueShortName] || 'No Long Name Provided'
```

Comment:
I found a discrepancy here regarding to including the Long Name wording is different, and is included in JSON compared to umm_tool_prototype; other than that it looks good.
<img width="1491" alt="Screenshot 2024-03-20 at 2 51 10 PM" src="https://github.com/nasa/mmt/assets/63017465/045d6b15-eec6-47fb-a529-cb98977309de">

<img width="1473" alt="Screenshot 2024-03-20 at 2 50 50 PM" src="https://github.com/nasa/mmt/assets/63017465/8c563484-8d4a-4b5d-a7c7-b70d09b45a5b">